src/_pytest/_code/code.py

```diff
@@ -396,8 +411,11 @@ def from_current(cls, exprinfo=None):
                          strip ``AssertionError`` from the output, defaults
                          to the exception message/``__str__()``
         """
-        tup = sys.exc_info()
-        assert tup[0] is not None, "no current exception"
+        tup_ = sys.exc_info()
+        assert tup_[0] is not None, "no current exception"
+        assert tup_[1] is not None, "no current exception"
```

Comment:
I forget which, but I seem to remember exception infos have the second tuple element as nullable

Reply:
I'll check that.

Reply:
In the cpython code, it looks like it's possible (at least there is no explicit check for that int the `sys.exc_info` implementation itself), but I don't think it can happen in practice. At least I couldn't find any reference to the exception itself being None but the exception type isn't.

Also, a patch to reflect that in typeshed was merged: https://github.com/python/typeshed/commit/4ae4714e00b167d4280b379cab66ea6f15b0c0d3